src/adapters/adapter.types.ts

```diff
@@ -8,6 +8,13 @@ export const ADAPTERS = Object.values(AdapterEnum);
 
 export type AdapterType = typeof AdapterEnum[keyof typeof AdapterEnum]
 
-export type Adapter = {
-  handleFile: (file: Stream) => unknown;
+export type BasicAdapterCommand<T> = {
+  file: Stream,
+  fileType: 'image' | 'video', // Info if processed media file is image or video
```

Comment:
```suggestion
  fileType: 'image' | 'video',
```
imo comment is not needed

Reply:
Can u create `FileType` enum and use it here?